extensions/vertx-http/runtime/src/main/java/io/quarkus/vertx/http/runtime/RouterProducer.java

```diff
@@ -10,9 +10,11 @@
 public class RouterProducer {
 
     private volatile Router router;
+    private volatile io.vertx.mutiny.ext.web.Router mutinyRouter;
 
-    void initialize(Router router) {
+    void initialize(Router router, io.vertx.mutiny.ext.web.Router mutinyRouter) {
```

Comment:
I think that we should get rid of the `RouterProducer` and replace the producer methods with synthetic beans. @cescoffier Would you like me to give it a try?

Reply:
Go ahead! I was actually surprised we still had producer methods.

Reply:
It seems that we would have to fix many cycles in the build chain :-(. So that's probably the reason ;-)

Reply:
Anyway, I'll try to fix it but let it go if there's no easy solution...